optimizely/odp/odp_manager.py

```diff
@@ -29,55 +29,59 @@
 class OdpManager:
     """Orchestrates segment manager, event manager and odp config."""
 
-    def __init__(self,
+    def __init__(
+        self,
         disable: bool,
-        cache_size: int,
-        cache_timeout_in_sec: int,
+        segments_cache: OptimizelySegmentsCache,
         segment_manager: Optional[OdpSegmentManager] = None,
         event_manager: Optional[OdpEventManager] = None,
-        logger: Optional[optimizely_logger.Logger] = None) -> None:
+        logger: Optional[optimizely_logger.Logger] = None
+    ) -> None:
 
         self.enabled = not disable
-        self.cache_size = cache_size
-        self.cache_timeout_in_sec = cache_timeout_in_sec
         self.odp_config = OdpConfig()
         self.logger = logger or optimizely_logger.NoOpLogger()
 
         self.segment_manager = None
         self.event_manager = None
 
         if not self.enabled:
+            self.logger.info('ODP is disabled.')
             return
 
         if segment_manager:
             segment_manager.odp_config = self.odp_config
             self.segment_manager = segment_manager
         else:
             self.segment_manager = OdpSegmentManager(self.odp_config,
```

Comment:
Can we consider removing odp_config from the constructor params?  It should be overriden for custom segment-manager anyway?  Let's talk about it when integrating at the top level.